### PR TITLE
[Metricbeat] Migrate Ceph cluster_disk to use ReporterV2 interface

### PR DESCRIPTION
**Pull request by sayden**
Refer to elastic/beats#10774 for more info

### CODE REVIEW

**Comment by sayden** [comment]
Error in Kibana/status module seems unrelated

**Comment by sayden** [comment]
https://github.com/elastic/beats/blob/master/metricbeat/module/ceph/cluster_disk/_meta/data.json does not need to be regenerated (in other words, when you regenerate it, nothing changes)

File: metricbeat/module/ceph/cluster_disk/cluster_disk_integration_test.go [hunk 1]
```diff
@@ -15,20 +15,31 @@
...
+	"github.com/stretchr/testify/assert"
+
 	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
 )
 
 func TestData(t *testing.T) {
-	f := mbtest.NewEventFetcher(t, getConfig())
-	err := mbtest.WriteEvent(f, t)
-	if err != nil {
+	t.Skip("Skipping data generation test")
```
Comment (ruflin) [inline metricbeat/module/ceph/cluster_disk/cluster_disk_integration_test.go]: Why do you skip it?

Comment (sayden) [inline metricbeat/module/ceph/cluster_disk/cluster_disk_integration_test.go]: CI test aren't launched with `-data` option but this is checked in `mbtest.WriteEventsReporterV2` in line 42. An error can still be raised in line 36 if ceph is not started by compose.

The problem is that Ceph start with Compose is buggy and it was omitted consistently in all Metricsets

Comment (sayden) [reply metricbeat/module/ceph/cluster_disk/cluster_disk_integration_test.go]: What a nice timing :smile: https://github.com/elastic/beats/pull/10990#discussion_r262405773 and more info here https://github.com/elastic/beats/pull/10993#discussion_r261627931

Comment (ruflin) [reply metricbeat/module/ceph/cluster_disk/cluster_disk_integration_test.go]: I see. As soon as https://github.com/elastic/beats/pull/10648 is merged we should switch over Ceph to generate the data file based on an http server as the current tests already use a http server.

Let's skip it for now but open a follow up issue to track:

* Implementation of data generation based on https://github.com/elastic/beats/pull/10648
* And investigation in how we can tests ceph against a "real" server like we do for all the other modules.

**Comment by sayden** [comment]
Pushed few things I saw on new commits, mainly https://github.com/elastic/beats/commit/30f809b18dbefa2cdc8f3df1d38f090a5b99e55d and https://github.com/elastic/beats/commit/45074a3efb74592505f90c9fc70e5dd75d898c1f

**Pull request merged**